### ISSUE TITLE
[Flaky test] ReceiverCannotReadWhenClosed

### ISSUE THREAD

**Issue by pakrym** [Azure/azure-sdk-for-net#17180]
```
Error message
Expected: instance of <Azure.Messaging.EventHubs.EventHubsException> and property Reason equal to ClientClosed
But was:  no exception thrown


Stack trace
   at Azure.Messaging.EventHubs.Tests.PartitionReceiverLiveTests.ReceiverCannotReadWhenClosed() in D:\a\1\s\sdk\eventhub\Azure.Messaging.EventHubs\tests\Primitives\PartitionReceiverLiveTests.cs:line 893
```

**Comment by jsquire** [comment]
Fix for this is part of the set in [#17115](https://github.com/Azure/azure-sdk-for-net/pull/17115) that'll be moved over once I uncover the root cause of the unrelated hang.

**Comment by jsquire** [comment]
There was a fix for this included in #17261
